### PR TITLE
Update branch enh6778 pull reqeust 318 with the latest from dev

### PR DESCRIPTION
**Pull request by ddimatos**
##### SUMMARY
This pull request is updating the branch enh6778 which has diverged over time from the dev branch. 

##### ISSUE TYPE
- Docs Pull Request

### CODE REVIEW

**Comment by ddimatos** [comment]
This pull request did introduce a few conflicts, from what i could see, they were pretty straight forward but to be sure nothing extraordinary was introduced I have included a patch of the diffs to make it easier to find. 
Files having conflicts were:
```
        plugins/module_utils/job.py
        plugins/modules/zos_job_output.py
        plugins/modules/zos_job_submit.py
        plugins/modules/zos_operator.py
```
Patch included:
[[job.py-zos,job_output.py,zos_job_submit.py,zos_operator.py].patch.zip](https://github.com/ansible-collections/ibm_zos_core/files/9032454/job.py-zos.job_output.py.zos_job_submit.py.zos_operator.py.patch.zip)

File: tests/functional/modules/test_zos_operator_func.py [hunk 1]
```diff
@@ -81,11 +81,21 @@ def test_zos_operator_positive_verbose_with_full_delay(ansible_zos_module):
     )
     endmod = time.time()
     timediff = endmod - startmod
-    assert timediff > 4
-    for result in results.contacted.values():
-        assert result["rc"] == 0
-        assert result.get("changed") is True
-        assert result.get("content") is not None
+    if timediff < 4:
+        print("\n........testable delay failed (short): assertion commented for testing....\n")
```
Comment (richp405) [inline tests/functional/modules/test_zos_operator_func.py:85]: I'm not sure, but I think this is wrong: "commented for testing"... That looks like one of my notes, but I don't recall working on this module.

**Review by richp405 — APPROVED** [review]
I left 2 comments: 1 about version requirements where we're skipping ansible 2.10, and one about a printout of a commented test.  Frankly, I don't remember putting it in, but I could see it might be needed until we validate the new time change test in zoau.  I'll approve, since it will pass tests, but thought you @ddimatos should note these two observations.

File: meta/runtime.yml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 ---
-requires_ansible: '>=2.9.6'
\ No newline at end of file
+requires_ansible: '>=2.9,>=2.11,<2.12'
```
[the remark this answers is not in the document]
Comment (ddimatos) [reply meta/runtime.yml:2]: You are right @richp405 , ill push that change after I get the work around for submitting a job done. That work around will allow for the dev branch to remain production ready and if ZOAU releases a fix we can do another change and update.

File: tests/functional/modules/test_zos_operator_func.py (hunk 1 above)
Comment (ddimatos) [reply tests/functional/modules/test_zos_operator_func.py:85]: @richp405 - i tracked it back to this change [here](https://github.com/ansible-collections/ibm_zos_core/blame/6bfe12c4ffd7a1243a2ebde590b10a8f20d3e8f2/tests/functional/modules/test_zos_operator_func.py) which was something you decided to do and I thought it was related to the timeout test, would have to go back into that PR and review again , screen shot included 
image

File: meta/runtime.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ddimatos) [reply meta/runtime.yml]: Updated in last commit

**Comment by ddimatos** [comment]
Pipeline has run regression successfully 
image
[Internal regression link](http://sprinkles1.fyre.ibm.com:8080/job/zosAnsible/992/)

**Comment by ddimatos** [comment]
After I performed my rubber ducky debug on all my changes, i decided to cite all the code needing to be updated after we obtain fixes for these from ZOAU. This will be easier to track because once merged into dev it wont be easy to see the code that needs to back out easily. 

[375](https://github.com/ansible-collections/ibm_zos_core/issues/375) [Enhancement] [Update the code to use a patched version of zoau jls when there is a fix](https://github.com/ansible-collections/ibm_zos_core/issues/375)
[376](https://github.com/ansible-collections/ibm_zos_core/issues/376) [[Enhancement] Python submit failing to honor hfs=true](https://github.com/ansible-collections/ibm_zos_core/issues/376)
as noted this is really 2 issues, the other is when `wait = false` returns NONE
377 - [[Enhancement] Validate timeout function delivered by ZOAU](https://github.com/ansible-collections/ibm_zos_core/issues/377)
378 - [[Enhancment] Evaluate and improve on the module security test case](https://github.com/ansible-collections/ibm_zos_core/issues/378)
379 - [[Enhancement] Enable all tests once the patch for querying operator responses is ready](https://github.com/ansible-collections/ibm_zos_core/issues/379)

**Review by richp405 — APPROVED** [review]
I don't see anything scary.

**Comment by ddimatos** [comment]
I am not going to delete branch `update/enh7668` so that we can reference if we need to for when backing out work around solutions. The work around changes are temporary and only being done this time around so that we can push to `dev` and continue to merge in all the other PRs sequentially behind this one. At one point we will correct the temporary changes once we are ready to test on a zoau beta.

**Pull request merged**